bucketlistapi/resources.py

```diff
@@ -0,0 +1,323 @@
+from collections import OrderedDict
+from flask import g, request
+from flask_httpauth import HTTPBasicAuth
+from flask_restful import abort, Resource
+from sqlalchemy import and_
+from webargs.flaskparser import use_args
+
+
+from bucketlistapi.utils import (user_reg_login_field, name_field,
+                                 name_done_field, limit_field, save)
+from bucketlistapi.models import BucketList, BucketListItem, User
+from bucketlistapi import db
+
+auth = HTTPBasicAuth()
+
+
+@auth.verify_password
+def verify_password(username_or_token, password):
+    # first try to authenticate by token
+    user = User.verify_auth_token(username_or_token)
+    if not user:
+        # try to authenticate with username/password
+        user = User.query.filter_by(username=username_or_token).first()
+        if not user or not user.verify_password(password):
+            return False
+    g.user = user
+    return True
+
+
+class UserRegAPI(Resource):
+    '''
+    The resource helps registers a new user with the post method.
+    [POST]
+
+    params: username and password.
+    '''
+
+    @use_args(user_reg_login_field)
+    def post(self, args):
+        # call methods to save user data and return success or otherwise
+        self.username = args.get('username')
+        self.password = args.get('password')
+        if self.check_user_exists():
+            abort(400, message='username already exists')
+        if (self.check_name() and
+                self.check_password()):
+            user = self.create_new_user()
+            return user.as_dict(), 201 if self.save_user(user) else abort(
+                400, message='bad request')
+        else:
+            abort(400, message='username/password minimum length is 8')
+
+    def check_name(self):
```

Comment:
I suggest changing this method's name to something that conveys it's purpose more clearly e.g name_is_valid because when calling it, self.name_is_valid is more readable than self.check_name. What are we checking?